src/pages/auth/signup/Signup.js

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import "./signup.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useReducer,useState } from "react";
+import { signupReducer } from "../../../reducer/signupReducer";
+import axios from "axios";
+import { useAuth } from "../../../contexts/Auth-context";
+
+export const Signup = () => {
+  const { navigate } = useNavigate();
+  const initialState = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  };
+
+  const [state, dispatch] = useReducer(signupReducer, initialState);
+  const {setUser} = useAuth()
+
+  const { firstName, lastName, email, password } = state;
+
+  const [showPass,setShowPass] = useState(false)
+
+  const signupSubmitHandler = async (e) => {
+    e.preventDefault();
+
+    try {
+      const response = await axios.post("/api/auth/signup",{
+        email,
+        firstName,
+        lastName,
+        password,
+
+    
+      })
+        console.log(response);
```

Comment:
remove console here

Reply:
> code structure is great . easy to read .one suggestion that work on UI little bit. Ur email password and signup should be on the left . so that its look good

Thank you ! i will do this in next commit